sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3718,6 +3718,14 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-33677: LikeSimplification should be skipped if escape is a wildcard character") {
+    val e = intercept[AnalysisException] {
+      sql("SELECT string(rand()) LIKE 'm%aca' ESCAPE '%'").collect()
```

Comment:
so the problem is, we should only apply like simplification if the pattern is valid?

Reply:
This rule may not work even for the case that has a valid pattern: `SELECT s LIKE 'maacaa' ESCAPE 'a' FROM t`